book.rb

```diff
@@ -0,0 +1,17 @@
+require_relative './item'
+
+class Book < Item
+  attr_accessor :publisher, :cover_state
+
+  def initialize(publish_date, archived, id, publisher, cover_state)
```

Comment:
Remember that the `id` parameter in the **Item** class is optional, so is necessary to make the same with the **Book** class' id.
You can change it to this:
```suggestion
  def initialize(publish_date, archived, publisher, cover_state, id = Time.now.to_f.to_s)
```
It will leave the id as the current time but with float format in string type so that it will never be the same as before.